### ISSUE TITLE
Multiple bundles with FuseBox.devServer

### ISSUE THREAD

**Issue by etienne-dldc** [fuse-box/fuse-box#134]
Hi (again),

## Some context

I have run FuseBox on a big TypeScript/React app I'm working on (just to see how hard it would be to use it instead of Webpack) and I get some illegal token error in certain .ts files.
The app is quite huge but we don't need to load everything every time, so we have used Webpack bundle split tools (`System.import`), alongside with wildcard require (`webpack.context`) to split our app in multiple bundles, then we have made a custom system to load dependencies from other bundles.

## What I managed to do with FuseBox

In the few hours I have played with FuseBox I have managed to create multiple bundles (`fusebox.bundle({ 'out': 'entry', ... })`) and load them individually with `FuseBox.import('file', cb)`.
Once a bundle is loaded, it's modules are accessible via (`FuseBox.import('default/...')`).

## What I do not have

From what I have seen in the source code, `FuseBox.devServer` does not accept an object as src just like `FuseBox.bundle` does and `FuseBox.bundle` does not have an option to watch files and rebuild on change.
This mean I can't run the server with all the files and I also can't run the bundler in watch mode.

## Possible solutions

I think the best would be to make `FuseBox.devServer` accept the same object type argument as `FuseBox.bundle` do.  
Another solution would be to add a `watch` option `FuseBox.bundle`.

## Bonus feature

Making Hot Reloading working with all that would be amazing :)

PS: Sorry for my English :/

**Comment by nchanged** [comment]
@etienne-dldc thanks for your issue. Valid points you have. Let's see what we can do.

Technically it should be possible for `devServer` to accept multi-bundle. But the problem is, that we need to understand what we are going to [watch](https://github.com/fuse-box/fuse-box/blob/master/src/devServer/Server.ts#L72) in this case.

**Comment by etienne-dldc** [comment]
In fact I think a good solution would be to move the `watch` part in `bundle()` or to create another method like `watch()`that would works just like Webpack does (bundle, watch and re-bundle on chnage).

This way it would be possible to watch and rebuild without the server and also it will make the devServer only a devServer and not a watcher + devServer.

For the "what we are going to watch" question, for me the correct behavior is to scope the watch option to the bundle but the consequence is that a `watch()` method would make no sense because it would mean that watch is on or off for all bundles.

## API change proposal

Here is a proposal that solve the problems above : 

### 1. Move `entry` to the config

First thing is to move the `entry` into the config passed to `init()`. I thing it would be a good thing to replace the string by an object (because it's more explicit) but that's another debate...

### 2. Add a watch option

Still on the config passed to `init()` we could add a watch option that would do what you expect to : watch and rebuild en change.

With all that we should get, for a single bundle projet something like this:

```
const fsbx = require("fuse-box");

let fuseBox = new fsbx.FuseBox({
    homeDir: "src/",
    entry: {
        path: "index.jsx",
        execute: true
    },
    watch: true,
    sourceMap: {
        bundleReference: "sourcemaps.js.map",
        outFile: "./build/sourcemaps.js.map",
    },
    outFile: "./build/out.js",
    plugins: [...]
});

fuseBox.bundle();
// or
fuseBox.devServer();
```

### `bundle()` and `devServer()` can take an array as parameter

`bundle()` and `devServer()` would take an array of object, and these object would have the same type as the option passed to init.

The result would be equivalent to having a default config (in `init()`) and override it for each bundle :

```
const fsbx = require("fuse-box");

let fuseBox = new fsbx.FuseBox({
    homeDir: "src/",
    sourceMap: {
        bundleReference: "sourcemaps.js.map",
        outFile: "./build/sourcemaps.js.map",
    },
    watch: true,
    outFile: "./build/out.js",
    plugins: [...]
});

fuseBox.bundle([
    // First bundle
    {
        entry: {
            path: "index.jsx",
            execute: true
        }
    },
    // Second bundle, don't execute and don't watch
    {
        entry: {
            path: "other-entry.jsx"
        },
        watch: false,
    },
]);
```

What do you think of this ?

**Comment by nchanged** [comment]
It looks a little bit webpackish, however, might work as an alternative to arithmetic. Let's look further into it. Thanks for the suggestion!

**Comment by alloy** [comment]
I too ran into this limitation, we have bundles for each distinct page.

I’m probably missing an important detail here, but why wouldn’t you simply watch the `homeDir` specified in `init`?

**Comment by nchanged** [comment]
That's it does at the moment. `devServer` should understand multiple bundles, i agree.

**Comment by nchanged** [comment]
Implemented in 1.4.1

**Comment by devmondo** [comment]
as long as this was Implemented in 1.4.1, we will close this. let us know otherwise :)
